src/nlp/api.jl

```diff
@@ -215,6 +387,114 @@ jprod!(
   Jv::AbstractVector,
 ) = jprod!(nlp, x, v, Jv)
 
+"""
+    Jv = lin_jprod(nlp, x, v)
+
+Evaluate ``J(x)v``, the linear Jacobian-vector product at `x`.
+"""
+function lin_jprod(nlp::AbstractNLPModel, x::AbstractVector, v::AbstractVector)
+  @lencheck nlp.meta.nvar x v
+  Jv = similar(v, nlp.meta.nlin)
+  return lin_jprod!(nlp, x, v, Jv)
+end
+
+"""
+    Jv = lin_jprod!(nlp, x, v, Jv)
+
+Evaluate ``J(x)v``, the linear Jacobian-vector product at `x` in place.
+"""
+function lin_jprod! end
+
+"""
+    Jv = lin_jprod!(nlp, rows, cols, vals, v, Jv)
+
+Evaluate ``J(x)v``, the linear Jacobian-vector product, where the Jacobian is given by
+`(rows, cols, vals)` in triplet format.
+"""
+function lin_jprod!(
+  nlp::AbstractNLPModel,
+  rows::AbstractVector{<:Integer},
+  cols::AbstractVector{<:Integer},
+  vals::AbstractVector,
+  v::AbstractVector,
+  Jv::AbstractVector,
+)
+  @lencheck nlp.meta.lin_nnzj rows cols vals
+  @lencheck nlp.meta.nvar v
+  @lencheck nlp.meta.nlin Jv
+  increment!(nlp, :neval_jprod)
+  coo_prod!(rows, cols, vals, v, Jv)
+end
+
+"""
+    Jv = lin_jprod!(nlp, x, rows, cols, v, Jv)
```

Comment:
How do you compute a product without the values?

Reply:
Not entirely sure for this one... I guess the implementation of the function should compute it directly making some use of the structure.

The same exists with `jprod!(nlp, rows, cols, v, Jv)` and `nln_jprod!(nlp, rows, cols, v, Jv)`.

Reply:
After discussion, this will be removed by #382